pyfstat/helper_functions.py

```diff
@@ -41,48 +42,80 @@ def set_up_matplotlib_defaults():
     plt.rcParams["axes.formatter.useoffset"] = False
 
 
-def set_up_command_line_arguments():
-    """Parse global commandline arguments."""
-    parser = argparse.ArgumentParser()
-    parser.add_argument(
-        "-v",
-        "--verbose",
-        action="store_true",
-        help="Increase output verbosity [logging.DEBUG]",
-    )
-    parser.add_argument(
-        "-q",
-        "--quite",
-        action="store_true",
-        help="Decrease output verbosity [logging.WARNING]",
-    )
-    args, unknown = parser.parse_known_args()
+def set_up_logger(outdir=None, label="pyfstat", log_level="INFO"):
+    """Setup the logger.
 
-    logger = logging.getLogger()
-    stream_handler = logging.StreamHandler()
-    stream_handler.setFormatter(
-        logging.Formatter("%(asctime)s %(levelname)-8s: %(message)s", datefmt="%H:%M")
-    )
+    Based on the implementation in Nessai:
+    https://github.com/mj-will/nessai/blob/main/nessai/utils/logging.py
+
+    Parameters
+    ----------
+    outdir : str, optional
+        Path of to outdir directory.
```

Comment:
Is this actually doing something currently? We don't seem to actively support stream redirection to files through a separate parameter here. But then I see below there's some logic using this param if something is found in `logging.FileHandler`. So is the idea that if, and only if, the user has already preconfigured some file output before calling this, we will intercept it and redirect to our outdir...?

Reply:
Yes. We actually were using that functionality before in `BaseSearchClass`.

`set_up_logger` takes a logger (the thing that logs) and attaches two handlers (things that "record" the logs): A `StreamHandler`, which I just fixed to log into STDOUT, and a `FileHandler`(as before), which goes into a file.

Loggers set up by `set_up_logger` *always* print logs to STDOUT *and* a specified file.

Reply:
> Loggers set up by set_up_logger always print logs to STDOUT and a specified file.

Ah, I didn't know that!

But do I understand right that when a file handler was preconfigured, we'll "steal" it silently?

Reply:
As in we will be writing to the same file?
I... think so?

Handlers are additive and independent, so I'd assume that we'd just be wrottong to the same file with different loggin label, but I haven't tried that situation.

Reply:
Yes: Some transient examples set the filename manually.

Reply:
Ok figured out an easy way to test: adding `logging.basicConfig(level=logging.INFO, filename="log.log")` in any example before a `pyfstat.set_up_logger` call -> all of the following get the info:
*    log.log
*    stdout
*    outdir/label.log

I think that's the expected behaviour. Thread resolved.